### PR TITLE
[MM-28060] Document Integrations team processes

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
I did consolidate the documentation from:
- Community server
- DMs
- Confluence

The PR is meant as a 1% draft. I mostly did copy and past existing test together.

https://github.com/jfrerich/plugin-release-process/blob/master/README.md should also get moved to the Handbook at some point. But that can happen in a different PR.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28060

### CODE REVIEW

**Comment by hanzei** [comment]
cc @mattermost/integrations

**Review by justinegeffen — DISMISSED** [review]

**Comment by hanzei** [comment]
@justinegeffen Heads up that I will cleanup the content of the Release process  this week. No need to touch it right now.

**Comment by justinegeffen** [comment]
Thanks, @hanzei! I'm marking this as WIP in case I forget. :)

**Comment by hanzei** [comment]
@justinegeffen I've moved the Plugin release docs into a separate document as it's not only interesting  for our team. PTAL.

File: operations/research-and-development/engineering/plugin-release-process.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+* Feature or patch bumping determined by commits being added from previous release tag.
+* Look through existing Issues and PRs and make sure the Milestone label is added for items to be included with release.
+* Verify no existing security issues (using [Updating Security Alerts Through CLI](#updating-security-alerts-through-cli)).
+  * If security issues exist, submit PR and merge before bumping version.
+* Install/Update [`pluginops`](https://github.com/mattermost/mattermost-utilities/tree/master/pluginops).
+* Run `pluginops release` inside a local copy of the repository.
+* Follow the steps of the tool.
+* Once the local changes are done, follow the link to open a PR.
+* Add any further reasoning or description for version bump (if necessary).
+* Add 2 Developers 1 QA for review.
```
Comment (justinegeffen) [inline operations/research-and-development/engineering/plugin-release-process.md]:
```suggestion
* Add 2 Developers and 1 QA for review.
```

File: operations/research-and-development/engineering/plugin-release-process.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+* Create a new branch so you can modify the plugin versions
+  * `git checkout -b bundle-plugins-v5.20`
+  * Use branch naming convention `bundle-plugins-vX.XX`
+* Edit `Makefile`
+  * Locate `# Plugins Packages` comment
+  * Modify plugin release versions
+* Create PR against master branch with following:
+  * **Title:** `Update bundled plugins for vX.XX`
+  * **Summary:** List of updated plugins
+    * Ideally includes from version -> to version for each plugin
```
Comment (justinegeffen) [inline operations/research-and-development/engineering/plugin-release-process.md]:
```suggestion
    * Ideally includes `from version` > `to version` for each plugin
```

File: operations/research-and-development/engineering/plugin-release-process.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+### Publish a plugin release version to the Plugin Marketplace
+
+The steps to have a plugin version added the Plugin Marketplace are included with the success of an `/mb cutplugin` slash command.
+
+### Release to `community.mattermost.com`
+
+Once the Marketplace PR is merged, request an update on the plugin in the `~community-configuration`[https://community.mattermost.com/core/channels/community-configuration] channel.
```
Comment (justinegeffen) [inline operations/research-and-development/engineering/plugin-release-process.md]:
```suggestion
Once the Plugin Marketplace PR is merged, request an update on the plugin in the `~community-configuration`[https://community.mattermost.com/core/channels/community-configuration] channel.
```

File: operations/research-and-development/engineering/plugin-release-process.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+## Security upgrade process
+
+(**`TODO`**: Automate checking all released plugins through CLI, cron, or GH webhook event)  
+(**`TODO`**: User must be repo admin to see and resolve automated security issue)  
+(**`TODO`**: Need method to hook to tell us when security issue is found)  
+(**`TODO`**: PR for security updates should be discrete)  
+(**`TODO`**: investigate `npm ls hoek`)
```
Comment (justinegeffen) [inline operations/research-and-development/engineering/plugin-release-process.md]:
```suggestion
(**`TODO`**: investigate `npm ls hook`)
```
@hanzei, is this an accurate correction?

File: operations/research-and-development/engineering/plugin-release-process.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  * Title: `Update Dependencies` (Will automatically get set)
+  * Summary: Update dependencies
+
+### Updating Security Alerts Through CLI
+
+* `git checkout latest master`
+* `git checkout -b npm-audit-fix`
+* `cd webapp/`
+* `npm audit` - will return list of security issues
+* `npm audit fix` - updates ``package-lock.json`` dependencies
```
Comment (justinegeffen) [inline operations/research-and-development/engineering/plugin-release-process.md]:
```suggestion
* `npm audit fix` - updates `package-lock.json` dependencies
```

File: operations/research-and-development/engineering/plugin-release-process.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+* `git add package-lock.json`
+* `git commit -m "Update dependencies"`
+* `git push --set-upstream origin npm-audit-fix`
+* Create PR
+  * Title: `Update Dependencies` (Will automatically get set)
+  * Summary: &lt;library\_name&gt; &lt;from\_ver&gt; -&gt; &lt;to\_ver&gt;
```
Comment (justinegeffen) [inline operations/research-and-development/engineering/plugin-release-process.md]: Is the inclusion of the "&lt;" and "&gt;" correct here?

File: operations/research-and-development/engineering/plugin-release-process.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+### Updating Security Alerts Through GitHub
+
+Github displays security alerts when viewing a GitHub repo.
```
Comment (justinegeffen) [inline operations/research-and-development/engineering/plugin-release-process.md]:
```suggestion
GitHub displays security alerts when viewing a GitHub repo.
```

File: operations/research-and-development/engineering/plugin-release-process.md (hunk 6 above)
Comment (justinegeffen) [reply operations/research-and-development/engineering/plugin-release-process.md]: Ah, I see how it's meant to render - should it be:

<library\_name>; <from\_ver> -> <to\_ver>

Comment (hanzei) [reply operations/research-and-development/engineering/plugin-release-process.md]: Updated :+1:

**Review by justinegeffen — APPROVED** [review]

**Comment by hanzei** [comment]
@justinegeffen https://handbook.mattermost.com/operations/research-and-development/engineering doesn't seem to contain the new content. What did I mess up?

**Comment by justinegeffen** [comment]
Oh that's on me - I didn't add this to the SUMMARY.md file. Opening a new PR for it, thank you!

**Pull request merged**